core/src/main/java/com/google/cloud/sql/core/SqlAdminApiFetcher.java

```diff
@@ -96,23 +93,27 @@ private String generatePublicKeyCert(KeyPair keyPair) {
 
   InstanceData getInstanceData(
       CloudSqlInstanceName instanceName,
-      OAuth2Credentials credentials,
+      AccessTokenSupplier accessTokenSupplier,
       AuthType authType,
       ListeningScheduledExecutorService executor,
       ListenableFuture<KeyPair> keyPair)
       throws ExecutionException, InterruptedException {
 
+    ListenableFuture<Optional<AccessToken>> token =
+        executor.submit(
+            () -> accessTokenSupplier != null ? accessTokenSupplier.get() : Optional.empty());
```

Comment:
If we initialize this correctly, we can always do `accessTokenSupplier.get()` here, I think.